src/components/Popover/utils.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect, useCallback, useRef } from 'react';
+
+/**
+ * transition - Helper function to generate transition properties rule
+ */
+
+type Transition = (properties: string[], duration: string, timingFunction: string) => string | null;
+
+const transition: Transition = (properties, duration, timingFunction) =>
+    properties.map(property => `${property} ${duration} ${timingFunction}`).join(',');
+
+/**
+ * useStateWithTimeout - Manages state with a Timeout
+ */
+
+type Return<S> = [S, (arg0: S) => void, (arg0: S) => void, () => void];
+
+function useStateWithTimeout<S>(defaultValue: S, timeout?: number): Return<S> {
```

Comment:
I can't wrap my head around the need for a timeout? 😬

Reply:
-> as said above, opacity transition animation :)